lib/ansible/plugins/filter/ipaddr.py

```diff
@@ -767,6 +770,23 @@ def ipsubnet(value, query='', index='x'):
                     return str(value.supernet(query)[0])
                 except:
                     return False
+    elif querystr:
+        try:
+            vtype = ipaddr(query, 'type')
+            if vtype == 'address':
+                v = ipaddr(query, 'cidr')
+            elif vtype == 'network':
+                v = ipaddr(query, 'subnet')
+
+            query = netaddr.IPNetwork(v)
+        except:
```

Comment:
It is a bad practice to catch all exceptions wholesale. You mask different errors, which you should've processed correctly. Also this makes it almost impossible to debug misbehavior.

Reply:
Yes, I see that, but it's not a good approach.
From https://github.com/drkjam/netaddr/blob/ebf714601e8cd6234575422f876d4d579b202b1a/netaddr/ip/__init__.py#L938 it looks like `netaddr.IPNetwork(v)` may raise `AddrFormatError`.

But if you want to catch all `netaddr` library's thrown exceptions, you may do:
```python
from netaddr.core import AddrConversionError, AddrFormatError, NotRegisteredError
NETADDR_ERRORS = AddrConversionError, AddrFormatError, NotRegisteredError
...
try:
    ...your exceptional code here
except NETADDR_ERRORS as err:
    process_or_log(err)
    # optionally raise/pass/return False (whatever fits)
```

Reply:
I still insist on correct exception handling.